### PR TITLE
Helm: Add support for dedicated ruler query path

### PR DESCRIPTION
**Pull request by alex5517**
#### What this PR does

This PR provides the option to deploy a read path dedicated to the ruler.
This allows you to scale user/dashboards queries independently from the ruler usage.

Another PR that does the same but has no activity #5081 

Closes https://github.com/grafana/mimir/pull/5081

#### Checklist

- [x] Tests updated.
- [x] Documentation added.
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`.
- [x] [`about-versioning.md`](https://github.com/grafana/mimir/blob/main/docs/sources/mimir/configure/about-versioning.md) updated with experimental features.

### CODE REVIEW

**Comment by lasermoth** [comment]
@dimitarvdimitrov Sorry for the ping, I wonder if you would have capacity to review this?

**Comment by dimitarvdimitrov** [comment]
i'll take a look later this week; apologies for the delay. At a glance one thing that's missing is a test values file. [This is one example](https://github.com/grafana/mimir/blob/adb6380a09b8c6ef45ca31b8a6066f93ff231bc6/operations/helm/charts/mimir-distributed/ci/offline). After creating the file and running `make build-helm-tests` there will all the resources generated from the values file. You can commit and push them. [This doc](https://github.com/grafana/mimir/blob/be805d06192bf3f458b3e522c1b2b69f7f0171fe/docs/internal/contributing/contributing-to-helm-chart.md#updating-compiled-manifests) has more details on what those manifests are useful for

**Comment by alex5517** [comment]
@dimitarvdimitrov,

I added the tests as requested :)

File: operations/helm/charts/mimir-distributed/values.yaml [hunk 1]
```diff
@@ -1280,6 +1284,301 @@ ruler:
...
+ruler_querier:
+  replicas: 2
+
+  # -- [Experimental] Configure autoscaling via KEDA (https://keda.sh). This requires having
+  # KEDA already installed in the Kubernetes cluster. The metrics for scaling are read
+  # according to top-level kedaAutoscaling.prometheusAddress (defaulting to metamonitoring remote-write destination).
+  # Basic auth and extra HTTP headers from metaMonitoring are ignored, please use customHeaders.
+  # The remote URL is used even if metamonitoring is disabled.
+  # See https://github.com/grafana/mimir/issues/7367 for more details on how to migrate to autoscaled resources without disruptions.
+  kedaAutoscaling:
```
Comment (dimitarvdimitrov) [inline operations/helm/charts/mimir-distributed/values.yaml:1301]: oh that's great! Thanks for adding it!

File: operations/helm/charts/mimir-distributed/values.yaml [hunk 2]
```diff
@@ -1280,6 +1284,301 @@ ruler:
   # Set to 'null' to use the Jaeger client's default value
   jaegerReporterMaxQueueSize: 1000
 
+  # -- If set to true, a dedicated query path will be deployed for the ruler and operational mode will be set to use remote evaluation. https://grafana.com/docs/mimir/latest/references/architecture/components/ruler/#remote
+  # -- This is useful for isolating the ruler queries from other queriers (api/grafana).
+  remoteEvaluationDedicatedQueryPath: false
+
+# -- Only deployed if .Values.remoteEvaluation.enabled && .Values.remoteEvaluation.dedicatedQueryPath
```
Comment (dimitarvdimitrov) [inline operations/helm/charts/mimir-distributed/values.yaml:1291]: All the comments refer to `.Values.remoteEvaluation.enabled`. Can you replace them with `ruler.remoteEvaluationDedicatedQueryPath`? (`.Values` is an implementation detail)

**Review by dimitarvdimitrov — COMMENTED** [review]
nice work! Thank you for cleaning this up and adding autoscaling. I only left a single suggestion regarding the outdated comments from the original PR

File: operations/helm/charts/mimir-distributed/CHANGELOG.md [hunk 3]
```diff
@@ -29,6 +29,7 @@ Entries should include a reference to the Pull Request that introduced the chang
 
 ## main / unreleased
 
+* [FEATURE] Add support for a dedicated query path for the ruler. This allows for the isolation of ruler and user query paths. #7964
```
Comment (dimitarvdimitrov) [inline operations/helm/charts/mimir-distributed/CHANGELOG.md:32]:
```suggestion
* [FEATURE] Add support for a dedicated query path for the ruler. This allows for the isolation of ruler and user query paths. Enable it via `ruler.remoteEvaluationDedicatedQueryPath: true`. #7964
```

File: operations/helm/charts/mimir-distributed/values.yaml (hunk 2 above)
Comment (alex5517) [reply operations/helm/charts/mimir-distributed/values.yaml]: @dimitarvdimitrov - Fixed, thanks :)

**Review by dimitarvdimitrov — APPROVED** [review]
LGTM! Thank you for your work @alex5517!

**Comment by dimitarvdimitrov** [comment]
This change will be included in the weekly `5.4.0-weekly.292` and in the next stable release - `5.4.0` or `6.0.0`

**Pull request merged**